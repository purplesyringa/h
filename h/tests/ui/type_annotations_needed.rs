fn main() {
    let meow = h::map! {
        for _;
        0 => (),
    };

    let nya = h::map! {
        for (i32, _);
        (0, 1) => (),
    };

    let mrrp = h::map! {
        0 => (),
    };
}
