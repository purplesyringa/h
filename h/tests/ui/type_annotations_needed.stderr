error: type annotations needed
       specify the key type manually by elaborating the `for {integer};` annotation
       note: integer sizes need to be specified exactly
 --> tests/ui/type_annotations_needed.rs:2:16
  |
2 |       let meow = h::map! {
  |  ________________^
3 | |         for _;
4 | |         0 => (),
5 | |     };
  | |_____^
  |
  = note: this error originates in the macro `$crate::h_macros::map` which comes from the expansion of the macro `h::map` (in Nightly builds, run with -Z macro-backtrace for more info)

error: type annotations needed
       specify the key type manually by elaborating the `for (i32, {integer});` annotation
       note: integer sizes need to be specified exactly
  --> tests/ui/type_annotations_needed.rs:7:15
   |
7  |       let nya = h::map! {
   |  _______________^
8  | |         for (i32, _);
9  | |         (0, 1) => (),
10 | |     };
   | |_____^
   |
   = note: this error originates in the macro `$crate::h_macros::map` which comes from the expansion of the macro `h::map` (in Nightly builds, run with -Z macro-backtrace for more info)

error: type annotations needed
       specify the key type manually by prepending `for {integer};` to the macro input
       note: integer sizes need to be specified exactly
  --> tests/ui/type_annotations_needed.rs:12:16
   |
12 |       let mrrp = h::map! {
   |  ________________^
13 | |         0 => (),
14 | |     };
   | |_____^
   |
   = note: this error originates in the macro `$crate::h_macros::map` which comes from the expansion of the macro `h::map` (in Nightly builds, run with -Z macro-backtrace for more info)
